client/lecturer/app/engagement.directive.js

```diff
@@ -19,9 +19,32 @@
         function link(scope, element, attrs, engagementCanvas) {
             console.log('Ready (Engagement Canvas link)');
 
-            engagementCanvas.canvas = element[0];
+            engagementCanvas.width = element.width();
+            engagementCanvas.height = element.height();
+
+            engagementCanvas.canvas = document.createElement('canvas');
+            element.append(engagementCanvas.canvas);
             engagementCanvas.ctx = engagementCanvas.canvas.getContext('2d');
 
+            engagementCanvas.canvas.width = engagementCanvas.width;
+            engagementCanvas.canvas.height = engagementCanvas.height;
```

Comment:
I see that it might be confusing. My point of doing it like this is that the .canvas is just a canvas within the engagement-canvas div and thus shouldn't be the one who stores the dimensions. What might improve this is renaming .canvas to .backgroundCanvas, and maybe store the element in stead of the width and height. I.e:
```
engagementCanvas.backgroundCanvas.width = engagementCanvas.element.width()
```